src/UrlSearchParams.ts

```diff
@@ -0,0 +1,151 @@
+import { duplicate } from './lang';
+
+/**
+ * A object with string keys and string or string array values that describes a query string.
+ */
+type ParamList = { [key: string]: string | string[] };
+
+/**
+ * Parse a query string, returning a ParamList object.
```

Comment:
Parse -> Parses (use third-person present for function summaries, as if it is describing what the function does)